src/main/utils/shell-env.ts

```diff
@@ -0,0 +1,51 @@
+/**
+ * Copyright (c) 2021 OpenLens Authors
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy of
+ * this software and associated documentation files (the "Software"), to deal in
+ * the Software without restriction, including without limitation the rights to
+ * use, copy, modify, merge, publish, distribute, sublicense, and/or sell copies of
+ * the Software, and to permit persons to whom the Software is furnished to do so,
+ * subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS
+ * FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR
+ * COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER
+ * IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN
+ * CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+ */
+
+import shellEnvironment from "shell-env";
+import logger from "../logger";
+
+export interface EnvironmentVariables {
+  readonly [key: string]: string;
+}
+
+let shellSyncFailed = false;
+
+export async function shellEnv(shell?: string, timeout?: number) : Promise<EnvironmentVariables> {
+  let envVars = {};
+
+  if (!shellSyncFailed) {
+    try {
+      envVars = await Promise.race([
+        shellEnvironment(shell),
+        new Promise((_resolve, reject) => setTimeout(() => {
+          reject(new Error("Resolving shell environment is taking very long. Please review your shell configuration."));
+        }, timeout))
+      ]);
+    } catch (error) {
+      logger.error(`shellEnv: ${error}`);
+      shellSyncFailed = true;
+    }
+  } else {
+    logger.error("shellSync(): Resolving shell environment took too long. Please review your shell configuration.");
+  }
```

Comment:
@jakolehm is this line skipping on subsequent calls?

Reply:
I don't understand this question 🤔

Reply:
It was in response to this comment made by you:

> @jim-docker what if we just mark "shell env" as "invalid" during boot and skip shell-env within terminal if we see an invalid shell?